agent/proxycfg/state.go

```diff
@@ -311,6 +315,17 @@ func (s *state) run(ctx context.Context, snap *ConfigSnapshot) {
 	sendCh := make(chan struct{})
 	var coalesceTimer *time.Timer
 
+	scheduleUpdate := func() {
+		r := s.rateLimiter.Reserve()
+
+		coalesceTimer = time.AfterFunc(coalesceTimeout+r.Delay(), func() {
+			// This runs in another goroutine so we can't just do the send
+			// directly here as access to snap is racy. Instead, signal the main
+			// loop above.
+			sendCh <- struct{}{}
```

Comment:
⚠️ There's a (pre-existing) goroutine leak here, where if `ctx` is canceled when there's a pending update this goroutine will never halt.

The fix is simple, but for posterity, I'll add it in a follow-up PR once this one is merged.